test/unit/os/shell_spec.lua

```diff
@@ -69,4 +57,25 @@ describe('shell functions', function()
       eq(0, status)
     end)
   end)
+
+  describe('shell_build_argv', function()
+    setup(function()
+      shell.p_sxq = to_cstr('(')
+      shell.p_sxe = to_cstr('"&|<>()@^')
+    end)
+
+    it('applies shellxescape and shellxquote', function()
+      local argv = ffi.cast('char**',
+                        shell.shell_build_argv(to_cstr('echo &|<>()@^'), nil))
+      eq(ffi.string(argv[0]), '/bin/bash')
+      eq(ffi.string(argv[1]), '-c')
+      eq(ffi.string(argv[2]), '(echo ^&^|^<^>^(^)^@^^)')
+      eq(nil, argv[3])
+    end)
+
+    teardown(function()
```

Comment:
Please change this to a `after_each` (runs after each `it(...)`) instead of a `teardown` (runs only once per `describe(...)`).